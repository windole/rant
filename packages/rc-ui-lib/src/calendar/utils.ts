/* eslint-disable no-nested-ternary */
export function compareMonth(date1: Date, date2: Date): number {
  const year1 = date1.getFullYear();
  const year2 = date2.getFullYear();

  if (year1 === year2) {
    const month1 = date1.getMonth();
    const month2 = date2.getMonth();
    return month1 === month2 ? 0 : month1 > month2 ? 1 : -1;
  }

  return year1 > year2 ? 1 : -1;
}

export function compareDay(day1: Date, day2: Date): number {
  const compareMonthResult = compareMonth(day1, day2);

  if (compareMonthResult === 0) {
    const date1 = day1.getDate();
    const date2 = day2.getDate();
    return date1 === date2 ? 0 : date1 > date2 ? 1 : -1;
  }

  return compareMonthResult;
}

export const cloneDate = (date: Date): Date => new Date(date);

export const cloneDates = (dates: Date | Date[]): Date | Date[] =>
  Array.isArray(dates) ? dates.map(cloneDate) : cloneDate(dates);

export function getDayByOffset(date: Date, offset: number): Date {
  const cloned = cloneDate(date);
  cloned.setDate(cloned.getDate() + offset);
  return cloned;
}

export const getPrevDay = (date: Date): Date => getDayByOffset(date, -1);
export const getNextDay = (date: Date): Date => getDayByOffset(date, 1);
export const getToday = (): Date => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return today;
};

export function calcDateNum(date: [Date, Date]): number {
  const day1 = date[0].getTime();
  const day2 = date[1].getTime();
  return (day2 - day1) / (1000 * 60 * 60 * 24) + 1;
}

export const getMonthEndDay = (year: number, month: number): number =>
  32 - new Date(year, month - 1, 32).getDate();

const calendarLang = {
  end: '结束',
  start: '开始',
  title: '日期选择',
  confirm: '确定',
  startEnd: '开始/结束',
  weekdays: ['日', '一', '二', '三', '四', '五', '六'],
  monthTitle: (year: number, month: number) => `${year}年${month}月`,
  rangePrompt: (maxRange: number) => `最多选择 ${maxRange} 天`,
};

export const t = (key: string, ...args): string => {
  if (args.length) {
    return calendarLang[key](...args);
  }
  return calendarLang[key];
};

export const formatMonthTitle = (date: Date): string =>
  t('monthTitle', date.getFullYear(), date.getMonth() + 1);
