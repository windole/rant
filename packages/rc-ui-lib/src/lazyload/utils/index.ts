import { ListenEvent } from '../PropsType';

/* eslint-disable prefer-rest-params */
const inBrowser = typeof window !== 'undefined';

export const DEFAULT_URL =
  'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';

// 需要监听的事件
export const DEFAULT_EVENTS: ListenEvent[] = [
  'scroll',
  'wheel',
  'resize',
  'mousewheel',
  'animationend',
  'transitionend',
  'touchmove',
];

export const hasIntersectionObserver =
  inBrowser &&
  'IntersectionObserver' in window &&
  'IntersectionObserverEntry' in window &&
  'intersectionRatio' in window.IntersectionObserverEntry.prototype;

export const modeType = {
  event: 'event',
  observer: 'observer',
};

export const call = (fn): void => fn && fn();

export function setImage(
  element: HTMLMediaElement | HTMLIFrameElement | HTMLImageElement | HTMLElement,
  imagePath: string,
): void {
  if (
    element instanceof HTMLIFrameElement ||
    element instanceof HTMLImageElement ||
    element instanceof HTMLMediaElement
  ) {
    element.src = imagePath;
  } else {
    element.style.backgroundImage = `url('${imagePath}')`;
  }
}

export function loadImage(imagePath: string): Promise<string> {
  return new Promise((resolve, reject) => {
    const img = new Image();
    img.src = imagePath;
    img.onload = () => resolve(imagePath);
    img.onerror = (err) => reject(err);
  });
}

export function addCssClassName(
  element: HTMLImageElement | HTMLElement,
  cssClassName: string,
): void {
  if (!element.className.includes(cssClassName)) {
    element.className += ` ${cssClassName}`;
  }
}

export function throttle(fn, threshold = 250, scope?) {
  let last;
  let deferTimer;
  return () => {
    // @ts-ignore
    const context = scope || this;

    const now = +new Date();
    const args = arguments;
    if (last && now < last + threshold) {
      // hold on to it
      clearTimeout(deferTimer);
      deferTimer = setTimeout(() => {
        last = now;
        fn.apply(context, args);
      }, threshold);
    } else {
      last = now;
      fn.apply(context, args);
    }
  };
}

export function debounce(func, wait, immediate?) {
  let timeout;
  let args;
  let context;
  let timestamp;
  let result;

  const later = function later() {
    const last = +new Date() - timestamp;

    if (last < wait && last >= 0) {
      timeout = setTimeout(later, wait - last);
    } else {
      timeout = null;
      if (!immediate) {
        result = func.apply(context, args);
        if (!timeout) {
          context = null;
          args = null;
        }
      }
    }
  };

  return function debounced() {
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    context = this;
    args = arguments;
    timestamp = +new Date();

    const callNow = immediate && !timeout;
    if (!timeout) {
      timeout = setTimeout(later, wait);
    }

    if (callNow) {
      result = func.apply(context, args);
      context = null;
      args = null;
    }

    return result;
  };
}
