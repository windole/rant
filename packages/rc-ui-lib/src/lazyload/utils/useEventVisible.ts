import { useEffect, useRef, useState } from 'react';
import { debounce as debounceFunc, DEFAULT_EVENTS, throttle as throttleFunc } from '.';
import { EventOptions } from '../PropsType';

// try to handle passive events
let passiveEventSupported = false;
try {
  const opts = Object.defineProperty({}, 'passive', {
    get() {
      passiveEventSupported = true;
    },
  });
  window.addEventListener('test', null, opts);
  // eslint-disable-next-line no-empty
} catch (e) {}

const passiveEvent = passiveEventSupported ? { capture: false, passive: true } : false;

const useEventVisible = (
  eventOptions: EventOptions,
): [React.MutableRefObject<HTMLDivElement | HTMLImageElement>, boolean] => {
  const {
    listenEvents = DEFAULT_EVENTS,
    offset,
    scrollContainer,
    debounce,
    throttle,
  } = eventOptions;
  // 是否可见
  const [visible, setVisible] = useState(false);
  // // 包裹容器
  const containerRef = useRef<HTMLDivElement | HTMLImageElement>(null);

  useEffect(() => {
    // 检查函数
    let checkVisible = () => {
      // 如果容器不存在则不计算
      if (!containerRef.current) return;
      // 获取当前组件位置
      const { top, left, width, height } = containerRef.current?.getBoundingClientRect();
      // 获取滚动容器位置
      const {
        top: parentTop,
        left: parentLeft,
        width: parentWidth,
        height: parentHeight,
      } = scrollContainer.getBoundingClientRect();
      // 计算屏幕高度和宽度
      const windowInnerHeight = window.innerHeight || document.documentElement.clientHeight;
      const windowInnerWidth = window.innerWidth || document.documentElement.clientWidth;
      // 计算滚动容器和屏幕可视区的相交区域
      const intersectionTop = Math.min(parentTop, 0);
      const intersectionLeft = Math.min(parentLeft, 0);

      const intersectionHeight =
        Math.min(windowInnerHeight, parentTop + parentHeight) - intersectionTop;
      const intersectionWidth =
        Math.min(windowInnerWidth, parentLeft + parentWidth) - intersectionLeft;
      // 计算组件距离可视区的高度
      const offsetTop = top - intersectionTop;
      const offsetLeft = left - intersectionLeft;
      // 偏移量计算[top, left]
      const offsets = Array.isArray(offset) ? offset : [offset, offset];
      // 通过上述距离判断组件是否在可见区域
      const isVisible =
        offsetTop - offsets[0] <= intersectionHeight &&
        offsetTop + height + offsets[0] >= 0 &&
        offsetLeft - offsets[1] <= intersectionWidth &&
        offsetLeft + width + offsets[1] >= 0;

      setVisible(isVisible);
      // 如果可见取消当前组件的所有的监听
      if (isVisible) {
        listenEvents.forEach((event) => {
          scrollContainer?.removeEventListener(event, checkVisible);
        });
      }
    };

    // 使用防抖节流增强checkVisible函数，这里优先防抖，其次节流
    if (debounce) {
      // 防抖函数
      checkVisible = debounceFunc(checkVisible, debounce);
    } else if (throttle) {
      // 节流函数
      checkVisible = throttleFunc(checkVisible, throttle);
    }
    // 批量监听事件
    listenEvents.forEach((event) => {
      scrollContainer?.addEventListener(event, checkVisible, passiveEvent);
    });

    // 自动执行一次
    checkVisible();

    return () => {
      listenEvents.forEach((event) => {
        scrollContainer?.removeEventListener(event, checkVisible, passiveEvent);
      });
    };
  }, [scrollContainer]);
  return [containerRef, visible];
};

export default useEventVisible;
