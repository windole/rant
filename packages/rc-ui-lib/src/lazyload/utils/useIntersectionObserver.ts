import { useCallback, useEffect, useState } from 'react';

const useIntersectionObserver = (
  observerOptions: IntersectionObserverInit,
  handleIntersection: IntersectionObserverCallback,
): ((node: Element) => void) => {
  const [element, setElement] = useState<Element>(null);

  const handleElement = useCallback((node: Element) => {
    setElement(node);
  }, []);

  useEffect(() => {
    if (!element) return;
    const observer = new IntersectionObserver(handleIntersection, observerOptions);
    observer.observe(element);
    // eslint-disable-next-line consistent-return
    return () => observer && observer.disconnect && observer.disconnect();
  }, [handleIntersection, observerOptions, element]);
  return handleElement;
};

export default useIntersectionObserver;
