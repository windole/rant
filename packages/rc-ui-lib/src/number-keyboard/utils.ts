import { KeyConfig, NumberKeyboardProps } from './PropsType';

const genBasicKeys = (props: NumberKeyboardProps) => {
  const keys: KeyConfig[] = Array(9)
    .fill('')
    .map((_, i) => ({ text: i + 1 }));

  if (props.randomKeyOrder) {
    keys.sort(() => (Math.random() > 0.5 ? 1 : -1));
  }

  return keys;
};

export const genDefaultKeys = (props: NumberKeyboardProps): KeyConfig[] => [
  ...genBasicKeys(props),
  { text: props.extraKey as string, type: 'extra' },
  { text: 0 },
  {
    text: props.showDeleteKey ? props.deleteButtonText : '',
    type: props.showDeleteKey ? 'delete' : '',
  },
];

export const genCustomKeys = (props: NumberKeyboardProps) => {
  const keys = genBasicKeys(props);
  const { extraKey } = props;
  const extraKeys = Array.isArray(extraKey) ? extraKey : [extraKey];

  if (extraKeys.length === 1) {
    keys.push({ text: 0, wider: true }, { text: extraKeys[0], type: 'extra' });
  } else if (extraKeys.length === 2) {
    keys.push(
      { text: extraKeys[0], type: 'extra' },
      { text: 0 },
      { text: extraKeys[1], type: 'extra' },
    );
  }

  return keys;
};
