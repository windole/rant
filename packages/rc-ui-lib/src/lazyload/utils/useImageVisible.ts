import { useCallback, useState } from 'react';
import { addCssClassName, loadImage, setImage } from '.';
import { IntersectionObserverEntryType, ObserverOptions } from '../PropsType';
import useIntersectionObserver from './useIntersectionObserver';

const defaultObserverOptions = {
  threshold: [0, 1],
};

const useImageVisible = (
  observerOptions: ObserverOptions = defaultObserverOptions,
  errorImage: string,
): [(node: Element) => void, boolean] => {
  const [isLoaded, setIsLoaded] = useState(false);

  const handleIntersection = useCallback(
    (entries: IntersectionObserverEntryType[]) => {
      entries
        .filter((entry) => entry.isIntersecting)
        .forEach((entry) => {
          const target = entry.target as HTMLImageElement | HTMLElement;
          const url = target.dataset.src || '';
          loadImage(url)
            .catch(() => {
              setImage(target, errorImage);
              addCssClassName(target, 'rc-lazyload__error');
              target.setAttribute('lazy', 'error');
            })
            .then((imagePath: string) => {
              setImage(target, imagePath);
              setIsLoaded(true);
              addCssClassName(target, 'rc-lazyload__loaded');
              target.setAttribute('lazy', 'loaded');
              target.removeAttribute('data-src');
            });
        });
    },
    [setIsLoaded, isLoaded],
  );
  const ref = useIntersectionObserver(observerOptions, handleIntersection);
  return [ref, isLoaded];
};

export default useImageVisible;
