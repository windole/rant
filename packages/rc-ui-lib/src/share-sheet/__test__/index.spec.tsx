import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { sleep } from '../../../tests/utils';
import { ShareSheet } from '..';

const optionsMuli = [
  [
    { name: '微信', icon: 'wechat' },
    { name: '朋友圈', icon: 'wechat-moments' },
    { name: '微博', icon: 'https://img.yzcdn.cn/vant/share-sheet-weibo.png' },
    { name: 'QQ', icon: 'qq' },
  ],
  [
    { name: '复制链接', icon: 'link' },
    { name: '分享海报', icon: 'poster' },
    { name: '二维码', icon: 'qrcode' },
    { name: '小程序码', icon: 'weapp-qrcode' },
  ],
];

describe('ShareSheet', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
  });

  test('should render cancel text when using cancel-text prop', async () => {
    wrapper = mount(<ShareSheet visible cancelText="foo" />);

    expect(toJson(wrapper.find('.rc-share-sheet__cancel'))).toMatchSnapshot();

    await wrapper.setProps({ cancelText: '' });
    expect(wrapper.find('.rc-share-sheet__cancel').exists()).toBeFalsy();
  });

  test('should render description and title when using description prop and title prop', async () => {
    wrapper = mount(<ShareSheet visible description="foo" title="hoo" />);

    expect(wrapper.find('.rc-share-sheet__description').html()).toMatchSnapshot();
    expect(wrapper.find('.rc-share-sheet__title').html()).toMatchSnapshot();

    await wrapper.setProps({ description: '' });
    expect(wrapper.find('.rc-share-sheet__description').exists()).toBeFalsy();
  });

  test('should allow to custom the className of option', () => {
    wrapper = mount(
      <ShareSheet
        visible
        options={[{ name: 'Link', icon: 'link', className: 'foo', description: 'hoo' }]}
      />,
    );

    const option = wrapper.find('.rc-share-sheet__option');

    expect(option.hasClass('foo')).toBeTruthy();
  });
  // description

  test('should render multi Line when option item is array', () => {
    wrapper = mount(<ShareSheet visible options={optionsMuli} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should emit select event when an option is clicked', () => {
    const onSelect = jest.fn();
    wrapper = mount(
      <ShareSheet visible options={[{ icon: 'wechat', name: 'wechat' }]} onSelect={onSelect} />,
    );

    wrapper.find('.rc-share-sheet__option').at(0).simulate('click');
    expect(onSelect).toHaveBeenCalledWith({ icon: 'wechat', name: 'wechat' }, 0);
  });

  test('should emit cancel event when the cancel button is clicked', () => {
    const onCancel = jest.fn();
    wrapper = mount(<ShareSheet visible onCancel={onCancel} />);

    wrapper.find('.rc-share-sheet__cancel').simulate('click');

    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  test('should emit click-overlay event when overlay is clicked', async () => {
    const onCancel = jest.fn();
    wrapper = mount(<ShareSheet visible cancelText="foo" onCancel={onCancel} />);

    await sleep();

    const overlay = wrapper.find('.rc-overlay')!;
    overlay.simulate('click');
    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  test('should have "van-popup--round" class when setting the round prop', async () => {
    wrapper = mount(<ShareSheet visible round />);

    expect(wrapper.find('.rc-popup--round').exists()).toBeTruthy();
  });
});
