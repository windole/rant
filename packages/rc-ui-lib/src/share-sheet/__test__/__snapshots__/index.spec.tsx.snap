// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareSheet should render cancel text when using cancel-text prop 1`] = `
<button
  className="rc-share-sheet__cancel"
  onClick={[Function]}
  type="button"
>
  foo
</button>
`;

exports[`ShareSheet should render description and title when using description prop and title prop 1`] = `"<div class=\\"rc-share-sheet__description\\">foo</div>"`;

exports[`ShareSheet should render description and title when using description prop and title prop 2`] = `"<div class=\\"rc-share-sheet__title\\">hoo</div>"`;

exports[`ShareSheet should render multi Line when option item is array 1`] = `
<ShareSheet
  cancelText="取消"
  closeOnClickOverlay={true}
  closeOnPopstate={true}
  options={
    Array [
      Array [
        Object {
          "icon": "wechat",
          "name": "微信",
        },
        Object {
          "icon": "wechat-moments",
          "name": "朋友圈",
        },
        Object {
          "icon": "https://img.yzcdn.cn/vant/share-sheet-weibo.png",
          "name": "微博",
        },
        Object {
          "icon": "qq",
          "name": "QQ",
        },
      ],
      Array [
        Object {
          "icon": "link",
          "name": "复制链接",
        },
        Object {
          "icon": "poster",
          "name": "分享海报",
        },
        Object {
          "icon": "qrcode",
          "name": "二维码",
        },
        Object {
          "icon": "weapp-qrcode",
          "name": "小程序码",
        },
      ],
    ]
  }
  safeAreaInsetBottom={true}
  visible={true}
>
  <Popup
    className="rc-share-sheet"
    closeIcon="cross"
    closeIconPosition="top-right"
    closeOnClickOverlay={true}
    closeOnPopstate={true}
    duration={300}
    lockScroll={true}
    onClose={[Function]}
    overlay={true}
    position="bottom"
    preventDefaultMouseDown={false}
    round={true}
    safeAreaInsetBottom={true}
    teleport={[Function]}
    visible={true}
  >
    <Portal
      containerInfo={
        <body
          class="rc-overflow-hidden"
        >
          <div
            class="rc-overlay"
            style="z-index: 2000; animation-duration: 300ms;"
          />
          <div
            class="rc-popup rc-popup--round rc-popup--bottom rc-safe-area-bottom rc-share-sheet"
            style="z-index: 2000; transition-duration: 300ms;"
          >
            <div
              class="rc-share-sheet__options"
            >
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-wechat.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  微信
                </span>
              </div>
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-wechat-moments.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  朋友圈
                </span>
              </div>
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-weibo.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  微博
                </span>
              </div>
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-qq.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  QQ
                </span>
              </div>
            </div>
            <div
              class="rc-share-sheet__options rc-share-sheet__options--border"
            >
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-link.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  复制链接
                </span>
              </div>
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-poster.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  分享海报
                </span>
              </div>
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-qrcode.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  二维码
                </span>
              </div>
              <div
                class="rc-share-sheet__option"
                role="button"
                tabindex="0"
              >
                <img
                  alt="share sheet icon"
                  class="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-weapp-qrcode.png"
                />
                <span
                  class="rc-share-sheet__name"
                >
                  小程序码
                </span>
              </div>
            </div>
            <button
              class="rc-share-sheet__cancel"
              type="button"
            >
              取消
            </button>
          </div>
        </body>
      }
    >
      <Overlay
        duration={300}
        lockScroll={true}
        onClick={[Function]}
        visible={true}
        zIndex={2000}
      >
        <CSSTransition2
          classNames="rc-fade"
          in={true}
          mountOnEnter={true}
          nodeRef={
            Object {
              "current": <div
                class="rc-overlay"
                style="z-index: 2000; animation-duration: 300ms;"
              />,
            }
          }
          timeout={300}
          unmountOnExit={true}
        >
          <Transition2
            appear={false}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            nodeRef={
              Object {
                "current": <div
                  class="rc-overlay"
                  style="z-index: 2000; animation-duration: 300ms;"
                />,
              }
            }
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={300}
            unmountOnExit={true}
          >
            <div
              className="rc-overlay"
              onClick={[Function]}
              style={
                Object {
                  "animationDuration": "300ms",
                  "zIndex": 2000,
                }
              }
            />
          </Transition2>
        </CSSTransition2>
      </Overlay>
      <CSSTransition2
        classNames="rc-popup-slide-bottom"
        in={true}
        mountOnEnter={true}
        nodeRef={
          Object {
            "current": <div
              class="rc-popup rc-popup--round rc-popup--bottom rc-safe-area-bottom rc-share-sheet"
              style="z-index: 2000; transition-duration: 300ms;"
            >
              <div
                class="rc-share-sheet__options"
              >
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-wechat.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    微信
                  </span>
                </div>
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-wechat-moments.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    朋友圈
                  </span>
                </div>
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-weibo.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    微博
                  </span>
                </div>
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-qq.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    QQ
                  </span>
                </div>
              </div>
              <div
                class="rc-share-sheet__options rc-share-sheet__options--border"
              >
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-link.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    复制链接
                  </span>
                </div>
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-poster.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    分享海报
                  </span>
                </div>
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-qrcode.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    二维码
                  </span>
                </div>
                <div
                  class="rc-share-sheet__option"
                  role="button"
                  tabindex="0"
                >
                  <img
                    alt="share sheet icon"
                    class="rc-share-sheet__icon"
                    src="https://img.yzcdn.cn/vant/share-sheet-weapp-qrcode.png"
                  />
                  <span
                    class="rc-share-sheet__name"
                  >
                    小程序码
                  </span>
                </div>
              </div>
              <button
                class="rc-share-sheet__cancel"
                type="button"
              >
                取消
              </button>
            </div>,
          }
        }
        onEnter={[Function]}
        onExited={[Function]}
        timeout={
          Object {
            "exit": 300,
          }
        }
      >
        <Transition2
          appear={false}
          enter={true}
          exit={true}
          in={true}
          mountOnEnter={true}
          nodeRef={
            Object {
              "current": <div
                class="rc-popup rc-popup--round rc-popup--bottom rc-safe-area-bottom rc-share-sheet"
                style="z-index: 2000; transition-duration: 300ms;"
              >
                <div
                  class="rc-share-sheet__options"
                >
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-wechat.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      微信
                    </span>
                  </div>
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-wechat-moments.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      朋友圈
                    </span>
                  </div>
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-weibo.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      微博
                    </span>
                  </div>
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-qq.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      QQ
                    </span>
                  </div>
                </div>
                <div
                  class="rc-share-sheet__options rc-share-sheet__options--border"
                >
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-link.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      复制链接
                    </span>
                  </div>
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-poster.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      分享海报
                    </span>
                  </div>
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-qrcode.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      二维码
                    </span>
                  </div>
                  <div
                    class="rc-share-sheet__option"
                    role="button"
                    tabindex="0"
                  >
                    <img
                      alt="share sheet icon"
                      class="rc-share-sheet__icon"
                      src="https://img.yzcdn.cn/vant/share-sheet-weapp-qrcode.png"
                    />
                    <span
                      class="rc-share-sheet__name"
                    >
                      小程序码
                    </span>
                  </div>
                </div>
                <button
                  class="rc-share-sheet__cancel"
                  type="button"
                >
                  取消
                </button>
              </div>,
            }
          }
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={
            Object {
              "exit": 300,
            }
          }
          unmountOnExit={false}
        >
          <div
            className="rc-popup rc-popup--round rc-popup--bottom rc-safe-area-bottom rc-share-sheet"
            onMouseDown={[Function]}
            style={
              Object {
                "display": undefined,
                "transitionDuration": "300ms",
                "zIndex": 2000,
              }
            }
          >
            <div
              className="rc-share-sheet__options"
              key="0"
            >
              <div
                className="rc-share-sheet__option"
                key="0"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-wechat.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  微信
                </span>
              </div>
              <div
                className="rc-share-sheet__option"
                key="1"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-wechat-moments.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  朋友圈
                </span>
              </div>
              <div
                className="rc-share-sheet__option"
                key="2"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-weibo.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  微博
                </span>
              </div>
              <div
                className="rc-share-sheet__option"
                key="3"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-qq.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  QQ
                </span>
              </div>
            </div>
            <div
              className="rc-share-sheet__options rc-share-sheet__options--border"
              key="1"
            >
              <div
                className="rc-share-sheet__option"
                key="0"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-link.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  复制链接
                </span>
              </div>
              <div
                className="rc-share-sheet__option"
                key="1"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-poster.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  分享海报
                </span>
              </div>
              <div
                className="rc-share-sheet__option"
                key="2"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-qrcode.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  二维码
                </span>
              </div>
              <div
                className="rc-share-sheet__option"
                key="3"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <img
                  alt="share sheet icon"
                  className="rc-share-sheet__icon"
                  src="https://img.yzcdn.cn/vant/share-sheet-weapp-qrcode.png"
                />
                <span
                  className="rc-share-sheet__name"
                >
                  小程序码
                </span>
              </div>
            </div>
            <button
              className="rc-share-sheet__cancel"
              onClick={[Function]}
              type="button"
            >
              取消
            </button>
          </div>
        </Transition2>
      </CSSTransition2>
    </Portal>
  </Popup>
</ShareSheet>
`;
